### PR TITLE
Add logic for pawn move

### PR DESCRIPTION
**Pull request by jmundraw**
Can someone check the logic for this? I didn't add anything about the piece being obstructed.  I also read online that the pawn can move 2 spaces if it is the first move for that pawn so I added logic for this.

### CODE REVIEW

File: app/models/pawn.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
 class Pawn < ChessPiece
 
+	def valid_move?(endpoint_x, endpoint_y)
+		if vertical_move?(endpoint_x, endpoint_y)
+			return true if position_y == 1 && endpoint_y <= 3
+			return true if endpoint_y == position_y + 1
```
Comment (varghes1) [inline app/models/pawn.rb]: that some solid coding logic!

**Comment by varghes1** [comment]
Looks good to merge!

**Comment by varghes1** [comment]
great job fixing the code. looks good to merge

File: app/models/pawn.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
 class Pawn < ChessPiece
 
+	def valid_move?(endpoint_x, endpoint_y)
+		if vertical_move?(endpoint_x, endpoint_y)
+			if self.color == "White"
+				return true if position_y == 1 && endpoint_y <= 3 || endpoint_y == position_y + 1
```
Comment (KaoruDev) [inline app/models/pawn.rb]: `return position_y == 1 && endpoint_y <= 3 || endpoint_y == position_y + 1`

**Pull request merged**